nixos/modules/services/web-apps/wordpress.nix

```
@@ -106,62 +106,36 @@ let
         };
 
         plugins = mkOption {
-          type = types.listOf types.path;
-          default = [];
+          type = types.attrsOf types.path;
+          default = {};
           description = lib.mdDoc ''
-            List of path(s) to respective plugin(s) which are copied from the 'plugins' directory.
+            Path(s) to respective plugin(s) which are copied from the 'plugins' directory.
 
             ::: {.note}
             These plugins need to be packaged before use, see example.
             :::
           '';
           example = literalExpression ''
-            let
-              # Wordpress plugin 'embed-pdf-viewer' installation example
-              embedPdfViewerPlugin = pkgs.stdenv.mkDerivation {
-                name = "embed-pdf-viewer-plugin";
-                # Download the theme from the wordpress site
-                src = pkgs.fetchurl {
-                  url = "https://downloads.wordpress.org/plugin/embed-pdf-viewer.2.0.3.zip";
-                  sha256 = "1rhba5h5fjlhy8p05zf0p14c9iagfh96y91r36ni0rmk6y891lyd";
-                };
-                # We need unzip to build this package
-                nativeBuildInputs = [ pkgs.unzip ];
-                # Installing simply means copying all files to the output directory
-                installPhase = "mkdir -p $out; cp -R * $out/";
-              };
-            # And then pass this theme to the themes list like this:
-            in [ embedPdfViewerPlugin ]
+            with pkgs.wordpressPackages.plugins; {
+              inherit embed-pdf-viewer-plugin;
+            }
```

```suggestion
            {
              inherit (pkgs.wordpressPackages.plugins) embed-pdf-viewer-plugin;
            }
```

It's trivial to avoid `with` here, so we should

---

(Please do the other one as well)

---

my thinking was that for users who are converting this:
```nix
with pkgs.wordpressPackages.plugins; [
  one
  two
  three
]
```
to the new format, that something like this would read better:
```nix
with pkgs.wordpressPackages.plugins; {
  inherit one;
  inherit two;
  inherit three;
}
```

---

> ```nix
> {
>   inherit (pkgs.wordpressPackages.plugins)
>     add-widget-after-content
>     hkdev-maintenance-mode
>     static-mail-sender-configurator;
> }
> ```